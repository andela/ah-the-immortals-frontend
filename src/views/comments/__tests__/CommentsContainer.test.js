import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import Comments from '../CommentsContainer';

describe('Tests CommentsContainer', () => {
  const store = configureStore([thunk])({
    comments: { 
      data: {comments: [{
        'id': 87,
        'createdAt': '2019-06-20',
        'updatedAt': '2019-06-20',
        'body': 'string',
        'author': {
          'username': 'slim',
          'bio': '',
          'image': 'hjn',
          'following': false
        },
        'parent': null,
        'likesInfo': {
          'like': false,
          'dislike': false,
          'like_count': 0,
          'dislikes_count': 0
        },
        'replies': [
          {
            'id': 87,
            'createdAt': '2019-06-20',
            'updatedAt': '2019-06-20',
            'body': 'string',
            'author': {
              'username': 'slim',
              'bio': '',
              'image': 'hjn',
              'following': false
            },
            'parent': null,
            'likesInfo': {
              'like': false,
              'dislike': false,
              'like_count': 0,
              'dislikes_count': 0
            }
          }]
      },]}, 
      error: {}}
  });
  const props = {
    data: {},
    getCommentsAction: jest.fn(),
    createCommentAction: jest.fn(),
    slug: 'string'
  };
  const history = createBrowserHistory();
  const wrapper = mount (
    <Router history={history}>
      <Provider store={store}>
        <Comments {...props} />
      </Provider>
    </Router>
  );
  it('tests mounting of container', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('tests input change in form', () => {
    const form = wrapper.find('form[className="form-horizontal"]');
    const input = form.find('textarea[name="body"]');
    input.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'Test body',
        name: 'body'
      }
    });
    expect(input.length).toEqual(1);
  });
  it('tests submit of a comment', () => {
    const form = wrapper.find('form[className="form-horizontal"]');
    form.simulate('submit', {
      preventDefault: jest.fn(),
    });
    expect(form.length).toEqual(1);
  });

});
