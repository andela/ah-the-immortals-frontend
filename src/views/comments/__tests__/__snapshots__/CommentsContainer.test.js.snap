// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CommentsContainer tests mounting of container 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Comments)
      createCommentAction={[MockFunction]}
      data={Object {}}
      getCommentsAction={[MockFunction]}
      slug="string"
    >
      <Comments
        createCommentAction={[Function]}
        data={
          Object {
            "comments": Array [
              Object {
                "author": Object {
                  "bio": "",
                  "following": false,
                  "image": "hjn",
                  "username": "slim",
                },
                "body": "string",
                "createdAt": "2019-06-20",
                "id": 87,
                "likesInfo": Object {
                  "dislike": false,
                  "dislikes_count": 0,
                  "like": false,
                  "like_count": 0,
                },
                "parent": null,
                "replies": Array [
                  Object {
                    "author": Object {
                      "bio": "",
                      "following": false,
                      "image": "hjn",
                      "username": "slim",
                    },
                    "body": "string",
                    "createdAt": "2019-06-20",
                    "id": 87,
                    "likesInfo": Object {
                      "dislike": false,
                      "dislikes_count": 0,
                      "like": false,
                      "like_count": 0,
                    },
                    "parent": null,
                    "updatedAt": "2019-06-20",
                  },
                ],
                "updatedAt": "2019-06-20",
              },
            ],
          }
        }
        getCommentsAction={[Function]}
        slug="string"
      >
        <div>
          <CreateComment
            onChange={[Function]}
            onSubmit={[Function]}
            slug="string"
            value=""
          >
            <div
              className="mb-5"
            >
              <h3>
                Leave your comment
              </h3>
              <form
                className="form-horizontal"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <textarea
                    className="form-control"
                    name="body"
                    onChange={[Function]}
                    placeholder="Leave your thoughts on this article here"
                    required={true}
                    rows="6"
                    value=""
                  />
                </div>
                <div
                  className="form-item"
                >
                  <div
                    className=""
                  >
                    <Button
                      active={false}
                      disabled={false}
                      type="submit"
                      variant="primary"
                    >
                      <button
                        className="btn btn-primary"
                        disabled={false}
                        type="submit"
                      >
                        ADD COMMENT
                      </button>
                    </Button>
                  </div>
                </div>
              </form>
              <hr />
            </div>
          </CreateComment>
          <div
            className="comment-tabs"
          >
            <ul
              className="nav nav-tabs"
              role="tablist"
            >
              <li
                className="nav-item active"
              >
                <a
                  className="nav-link active"
                  data-toggle="tab"
                  href="#comments-logout"
                  role="tab"
                >
                  <h5
                    className="text-capitalize"
                  >
                    Comments
                     
                    <span
                      className="comments-count"
                    >
                      1
                    </span>
                  </h5>
                </a>
              </li>
              <li
                className="nav-item"
              />
            </ul>
            <ViewComments
              comments={
                Array [
                  Object {
                    "author": Object {
                      "bio": "",
                      "following": false,
                      "image": "hjn",
                      "username": "slim",
                    },
                    "body": "string",
                    "createdAt": "2019-06-20",
                    "id": 87,
                    "likesInfo": Object {
                      "dislike": false,
                      "dislikes_count": 0,
                      "like": false,
                      "like_count": 0,
                    },
                    "parent": null,
                    "replies": Array [
                      Object {
                        "author": Object {
                          "bio": "",
                          "following": false,
                          "image": "hjn",
                          "username": "slim",
                        },
                        "body": "string",
                        "createdAt": "2019-06-20",
                        "id": 87,
                        "likesInfo": Object {
                          "dislike": false,
                          "dislikes_count": 0,
                          "like": false,
                          "like_count": 0,
                        },
                        "parent": null,
                        "updatedAt": "2019-06-20",
                      },
                    ],
                    "updatedAt": "2019-06-20",
                  },
                ]
              }
              slug="string"
            >
              <div
                className="tab-content"
                key="1"
              >
                <div
                  className="media-body ml-3"
                >
                  <div
                    className="card well"
                  >
                    <div
                      className="card-body bg-light"
                    >
                      <div
                        className="row mb-2"
                      >
                        <div
                          className="col-1"
                        >
                          <Link
                            to="/Profile/slim"
                          >
                            <a
                              href="/Profile/slim"
                              onClick={[Function]}
                            >
                              <img
                                alt="profile"
                                className="avatar"
                                src="hjn"
                              />
                            </a>
                          </Link>
                        </div>
                        <div
                          className="col-11 mt-2 mb-2"
                        >
                          <p
                            className="card-subtitle text-muted"
                          >
                            slim
                          </p>
                          <p
                            className="text-muted date-size"
                          >
                            12:00 am, Jun 20th 2019
                          </p>
                        </div>
                      </div>
                      <p
                        className="card-text"
                      >
                        string
                      </p>
                      <hr />
                      <Button
                        active={false}
                        className="card-link btn-comment"
                        data-target="#createreply-87"
                        data-toggle="collapse"
                        disabled={false}
                        type="button"
                        variant="outline-success"
                      >
                        <button
                          className="card-link btn-comment btn btn-outline-success"
                          data-target="#createreply-87"
                          data-toggle="collapse"
                          disabled={false}
                          type="button"
                        >
                          <i
                            className="fas fa-share"
                          />
                           
                          Reply
                        </button>
                      </Button>
                      <Button
                        active={false}
                        className="card-link btn-comment"
                        data-target="#replies-87"
                        data-toggle="collapse"
                        disabled={false}
                        type="button"
                        variant="outline-info"
                      >
                        <button
                          className="card-link btn-comment btn btn-outline-info"
                          data-target="#replies-87"
                          data-toggle="collapse"
                          disabled={false}
                          type="button"
                        >
                          <i
                            className="fas fa-comment-alt"
                          />
                           
                          <span
                            className="comments-count"
                          >
                            1
                          </span>
                           
                          replies
                        </button>
                      </Button>
                      <div
                        className="collapse mt-3"
                        id="createreply-87"
                      >
                        <Connect(Replies)
                          id={87}
                          slug="string"
                        >
                          <Replies
                            createChildCommentAction={[Function]}
                            data={
                              Object {
                                "comments": Array [
                                  Object {
                                    "author": Object {
                                      "bio": "",
                                      "following": false,
                                      "image": "hjn",
                                      "username": "slim",
                                    },
                                    "body": "string",
                                    "createdAt": "2019-06-20",
                                    "id": 87,
                                    "likesInfo": Object {
                                      "dislike": false,
                                      "dislikes_count": 0,
                                      "like": false,
                                      "like_count": 0,
                                    },
                                    "parent": null,
                                    "replies": Array [
                                      Object {
                                        "author": Object {
                                          "bio": "",
                                          "following": false,
                                          "image": "hjn",
                                          "username": "slim",
                                        },
                                        "body": "string",
                                        "createdAt": "2019-06-20",
                                        "id": 87,
                                        "likesInfo": Object {
                                          "dislike": false,
                                          "dislikes_count": 0,
                                          "like": false,
                                          "like_count": 0,
                                        },
                                        "parent": null,
                                        "updatedAt": "2019-06-20",
                                      },
                                    ],
                                    "updatedAt": "2019-06-20",
                                  },
                                ],
                              }
                            }
                            id={87}
                            slug="string"
                          >
                            <CreateReply
                              replySubmit={[Function]}
                              replyWrite={[Function]}
                              value=""
                            >
                              <Form
                                inline={false}
                                onSubmit={[Function]}
                              >
                                <form
                                  className=""
                                  onSubmit={[Function]}
                                >
                                  <Bootstrap(Row)>
                                    <Row
                                      bsPrefix="row"
                                      noGutters={false}
                                    >
                                      <div
                                        className="row"
                                      >
                                        <Col>
                                          <div
                                            className="col"
                                          >
                                            <FormControl
                                              className="input-width"
                                              name="body"
                                              onChange={[Function]}
                                              placeholder="Leave a reply"
                                              required={true}
                                              value=""
                                            >
                                              <input
                                                className="input-width form-control"
                                                name="body"
                                                onChange={[Function]}
                                                placeholder="Leave a reply"
                                                required={true}
                                                value=""
                                              />
                                            </FormControl>
                                          </div>
                                        </Col>
                                        <Col
                                          lg="2"
                                          xs={true}
                                        >
                                          <div
                                            className="col-lg-2 col"
                                          >
                                            <Button
                                              active={false}
                                              disabled={false}
                                              type="submit"
                                              variant="primary"
                                            >
                                              <button
                                                className="btn btn-primary"
                                                disabled={false}
                                                type="submit"
                                              >
                                                Send
                                              </button>
                                            </Button>
                                          </div>
                                        </Col>
                                      </div>
                                    </Row>
                                  </Bootstrap(Row)>
                                </form>
                              </Form>
                            </CreateReply>
                          </Replies>
                        </Connect(Replies)>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="collapse"
                  id="replies-87"
                >
                  <div
                    className="tab-pane"
                    key="1"
                  >
                    <div
                      className="media-body ml-5"
                    >
                      <div
                        className="card well"
                      >
                        <div
                          className="card-body bg-light"
                        >
                          <div
                            className="row mb-2"
                          >
                            <div
                              className="col-1"
                            >
                              <Link
                                to="slim"
                              >
                                <a
                                  href="/slim"
                                  onClick={[Function]}
                                >
                                  <img
                                    alt="profile"
                                    className="avatar"
                                    src="hjn"
                                  />
                                </a>
                              </Link>
                            </div>
                            <div
                              className="col-11 mt-2 mb-2"
                            >
                              <p
                                className="card-subtitle text-muted"
                              >
                                slim
                              </p>
                              <p
                                className="text-muted date-size"
                              >
                                12:00 am, Jun 20th 2019
                              </p>
                            </div>
                          </div>
                          <p
                            className="card-text"
                          >
                            string
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </ViewComments>
             
          </div>
        </div>
      </Comments>
    </Connect(Comments)>
  </Provider>
</Router>
`;
